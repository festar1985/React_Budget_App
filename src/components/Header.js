import React from "react";
import { NavLink } from "react-router-dom";
import { startLogOut } from "../actions/authentication";
import { connect } from "react-redux";

const Header = (props) => (
  <header>
    <h1> Budget APP</h1>

    <NavLink to="/dashboard" activeClassName="is-active" exact={true}>
      Dashboard
    </NavLink>
    <NavLink to="/create" activeClassName="is-active">
      Create
    </NavLink>
    <NavLink to="/help" activeClassName="is-active">
      Help
    </NavLink>
    <button onClick={props.signOut}>LogOut</button>
  </header>
);

const mapDispatchToProps = (dispatch) => ({
  signOut: () => {
    dispatch(startLogOut());
  },
});

export default connect(undefined, mapDispatchToProps)(Header);
