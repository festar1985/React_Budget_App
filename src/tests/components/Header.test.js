import React from "react";
import Header from "../../components/Header";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import Enzyme from "enzyme";

Enzyme.configure({ adapter: new Adapter() });

test("should render Header correclty", () => {
  const wrapper = shallow(<Header />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});
