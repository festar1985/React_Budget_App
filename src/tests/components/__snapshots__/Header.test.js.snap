// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correclty 1`] = `
<header>
  <h1>
     Budget APP
  </h1>
  <NavLink
    activeClassName="is-active"
    exact={true}
    to="/"
  >
    Homepage
  </NavLink>
  <NavLink
    activeClassName="is-active"
    to="/create"
  >
    Create
  </NavLink>
  <NavLink
    activeClassName="is-active"
    to="/help"
  >
    Help
  </NavLink>
</header>
`;
